server/server.go

```diff
@@ -538,3 +548,38 @@ func (s *Server) StartExitMonitor() {
 	}
 	<-done
 }
+
+// StartConfigWatcher starts a new watching go routine for the provided
+// `fileName` and `reloadFunc`. The method errors if the given fileName does
+// not exist or is not accessible.
+func (s *Server) StartConfigWatcher(
+	fileName string,
+	reloadFunc func(string) error,
+) (chan os.Signal, error) {
+	// Validate the arguments
+	if _, err := os.Stat(fileName); err != nil {
+		return nil, err
+	}
+	if reloadFunc == nil {
+		return nil, fmt.Errorf("provided reload closure is nil")
+	}
+
+	// Setup the signal notifier
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, signals.Hup)
+
+	go func() {
+		for {
+			// Block until the signal is received
+			<-c
+			logrus.Infof("reloading configuration %q", fileName)
+			if err := reloadFunc(fileName); err != nil {
+				logrus.Warnf("unable to reload configuration: %v", err)
```

Comment:
Can we make this an `Errorf()`? I think that's something to be screamed.

Reply:
Yes, I applied it.